src/lib/i18n/index.es

```diff
@@ -0,0 +1,211 @@
+/**
+ * Default language.
+ * @type {string}
+ */
+const DEFAULT_LANGUAGE = 'en'
+
+/**
+ * Pattern to detect params in a string.
+ * @type {Regexp}
+ */
+let paramsPattern = /%(\d+)/g
+
+/**
+ * To store all the created instances.
+ * @type {Object}
+ */
+let instances = {}
+
+/**
+ * Current language. Shared by all the instances.
+ * @type {String}
+ */
+let language = DEFAULT_LANGUAGE
+
+/**
+ * Public API.
+ * @type {Object}
+ */
+class I18N {
+
+  /**
+   * Creates an i18n instance.
+   * If the ID is specified returns a singleton for each ID.
+   * If the ID is not specified returns a new instance.
+   * @param {string} id Instance ID. Optional.
+   */
+  constructor (id) {
+    let isValidId = typeof id === 'string'
+
+    if (!!isValidId && !!instances[id]) {
```

Comment:
Well, because case "1" is the "native/default behavior" case.

If noone says the contrary an instance is created. There's no need to alter the default behavior of the constructor by forcing a "return this".
It's a "2 returns" vs "3 returns".

The early return in my case is if we want to "break" the default constructor behavior.